### DISCUSSION TITLE
Which platform_info.c should we refer to?

### DISCUSSION THREAD

**Discussion by sinall** [OpenAMP/open-amp#456] [Q&A]
We are currently experimentalling Open-AMP with [OpenHarmony LiteOS-A](https://gitee.com/openharmony/kernel_liteos_a/blob/master/README.md) on [3516DV300](https://www.hisilicon.com/en/products/smart-vision/consumer-camera/IOTVision/Hi3516DV300) board. We want to make as less efforts as possible to make the `echo` program run on LiteOS-A. LiteOS-A is similar to Linux, it has POSIX API layer.

What we are trying:
1. Run LiteOS-A with rpmsg-sample-ping on primary core using first 86MB physical memory
2. Run LiteOS-A with rpmsg-sample-echo on secondary core using middle 40MB physical memory
3. Map the last 2MB to /dev/uio0 for above 2 LiteOS-A

In Open-AMP, we find several different implementations of `platform_info.c`:
1. apps/system/linux/machine/generic/platform_info.c
2. apps/machine/zynqmp_r5/platform_info.c
3. apps/machine/zynqmp/platform_info.c
4. apps/machine/zynq7/platform_info.c

We have tried to adapt `apps/system/linux/machine/generic/platform_info.c` to `apps/system/liteos_a/machine/generic/platform_info.c`, but the `shm_open` is broken on LiteOS-A. Then we found the other implementations uses `metal_device_open` but `metal_shmem_open`.

Questions:
1. What's the relationship between `openamp.shm` and `/dev/uio0`?
2. Should we continue to try `metal_shmem_open`, or should we try to use `metal_device_open` like `zynqmp`?

**Comment by arnopo** [comment]
Not sure that the Linux implementation is adapted. I suggest you to have a look instead in https://github.com/OpenAMP/open-amp/blob/main/apps/machine/zynqmp_r5/platform_info.c. this implemntation is adapted for 
Some other implementations which do not use the OpenAMp remoteproc framework:
- Zephyr:
    - host: https://elixir.bootlin.com/zephyr/latest/source/samples/subsys/ipc/openamp/src/main.c
    - copro: https://elixir.bootlin.com/zephyr/latest/source/samples/subsys/ipc/openamp/remote/src/main.c
- ST baremetal copro (using resource table): https://elixir.bootlin.com/zephyr/latest/source/samples/subsys/ipc/openamp/src/main.c#L167
